exchanges/kucoin/kucoin_futures.go

```diff
@@ -0,0 +1,1087 @@
+	kucoinFuturesFundingHistory      = "/v1/funding-history"
+
+	kucoinFuturesAccountOverview              = "/v1/account-overview"
+	kucoinFuturesTransactionHistory           = "/v1/transaction-history"
+	kucoinFuturesSubAccountAPI                = "/v1/sub/api-key"
+	kucoinFuturesDepositAddress               = "/v1/deposit-address"
+	kucoinFuturesDepositsList                 = "/v1/deposit-list"
+	kucoinFuturesWithdrawalLimit              = "/v1/withdrawals/quotas"
+	kucoinFuturesWithdrawalList               = "/v1/withdrawal-list"
+	kucoinFuturesCancelWithdrawal             = "/v1/withdrawals/%s"
+	kucoinFuturesTransferFundtoMainAccount    = "/v3/transfer-out"
+	kucoinFuturesTransferFundtoFuturesAccount = "/v1/transfer-in"
+	kucoinFuturesTransferOutList              = "/v1/transfer-list"
+	kucoinFuturesCancelTransferOut            = "/v1/cancel/transfer-out"
+)
+
+// GetFuturesOpenContracts gets all open futures contract with its details
+func (ku *Kucoin) GetFuturesOpenContracts(ctx context.Context) ([]Contract, error) {
+	resp := struct {
+		Data []Contract `json:"data"`
+		Error
+	}{}
+
+	return resp.Data, ku.SendHTTPRequest(ctx, exchange.RestFutures, kucoinFuturesOpenContracts, &resp)
+}
+
+// GetFuturesContract get contract details
+func (ku *Kucoin) GetFuturesContract(ctx context.Context, symbol string) (Contract, error) {
+	resp := struct {
+		Data Contract `json:"data"`
+		Error
+	}{}
+
+	if symbol == "" {
+		return resp.Data, errors.New("symbol can't be empty")
+	}
+	return resp.Data, ku.SendHTTPRequest(ctx, exchange.RestFutures, fmt.Sprintf(kucoinFuturesContract, symbol), &resp)
+}
+
+// GetFuturesRealTimeTicker get real time ticker
+func (ku *Kucoin) GetFuturesRealTimeTicker(ctx context.Context, symbol string) (*FuturesTicker, error) {
+	resp := struct {
+		Data FuturesTicker `json:"data"`
+		Error
+	}{}
+
+	params := url.Values{}
+	if symbol == "" {
+		return &resp.Data, errors.New("symbol can't be empty")
+	}
+	params.Set("symbol", symbol)
+	return &resp.Data, ku.SendHTTPRequest(ctx, exchange.RestFutures, common.EncodeURLValues(kucoinFuturesRealTimeTicker, params), &resp)
+}
+
+// GetFuturesOrderbook gets full orderbook for a specified symbol
+func (ku *Kucoin) GetFuturesOrderbook(ctx context.Context, symbol string) (*Orderbook, error) {
+	var o futuresOrderbookResponse
+	params := url.Values{}
+	if symbol == "" {
+		return nil, errors.New("symbol can't be empty")
+	}
+	params.Set("symbol", symbol)
+	err := ku.SendHTTPRequest(ctx, exchange.RestFutures, common.EncodeURLValues(kucoinFuturesFullOrderbook, params), &o)
+	if err != nil {
+		return nil, err
+	}
+	return constructFuturesOrderbook(&o)
+}
+
+// GetFuturesPartOrderbook20 gets orderbook for a specified symbol with depth 20
+func (ku *Kucoin) GetFuturesPartOrderbook20(ctx context.Context, symbol string) (*Orderbook, error) {
+	var o futuresOrderbookResponse
+	params := url.Values{}
+	if symbol == "" {
+		return nil, errors.New("symbol can't be empty")
```

Comment:
So just to reiterate we should have a variable for errors. This is so we can call `errors.Is(err, expectedError)` in tests and exactly match to the error return. This helps in code coverage and would need to be implemented in the test files.